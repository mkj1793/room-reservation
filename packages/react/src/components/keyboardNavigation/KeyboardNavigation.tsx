import React from 'react';

import { KeyboardTrackerProps } from '../../hooks/keyboardNavigation/index';
import { useKeyboardNavigation } from '../../hooks/useKeyboardNavigation';

// import base styles
import '../../styles/base.css';

export type KeyboardNavigationProps = React.PropsWithChildren<KeyboardTrackerProps & { className?: string }>;

export const KeyboardNavigation = ({ children, className, ...rest }: KeyboardNavigationProps) => {
  const { ref } = useKeyboardNavigation(rest);
  return (
    <div className={className} ref={ref}>
      {children}
    </div>
  );
};
