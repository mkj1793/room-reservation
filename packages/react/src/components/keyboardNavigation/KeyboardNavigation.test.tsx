import React from 'react';
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';

import { KeyboardNavigation } from './KeyboardNavigation';

describe('<KeyboardNavigation /> spec', () => {
  it('renders the component', () => {
    const { asFragment } = render(<KeyboardNavigation />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should not have basic accessibility issues', async () => {
    const { container } = render(<KeyboardNavigation />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
