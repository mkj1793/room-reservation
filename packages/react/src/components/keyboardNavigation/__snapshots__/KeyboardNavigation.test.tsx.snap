// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<KeyboardNavigation /> spec renders the component 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
